### PR TITLE
Fixed #1147 Missing support for license attribute in Repository

### LINKED ISSUE #1147: Missing support for license attribute in Repository
There is a missing attribute for the license attribute which is returned when Repository API is called. At the moment there is only get_license function, but that makes another request.

I have a PR for this issue.

### PR DESCRIPTION
**Pull request by jppiiroinen**
This adds new attribute 'license' to Repository. Which allows access to the license information which is already retrieved earlier.

### CODE REVIEW

**Comment by jppiiroinen** [comment]
Noticed that I have not updated the unit tests. I do not have time right now to do that one, but I will contribute those to this PR a bit later.

File: github/License.py [hunk 1]
```diff
@@ -160,3 +161,11 @@ def _useAttributes(self, attributes):
             self._conditions = self._makeListOfStringsAttribute(attributes["conditions"])
         if "limitations" in attributes:  # pragma no branch
             self._limitations = self._makeListOfStringsAttribute(attributes["limitations"])
+
+
+class RepositoryLicense(License):
```
Comment (s-t-e-v-e-n-k) [inline github/License.py]: Part of me questions why you're adding this? If License perfectly encapsulates the data returned, I see no reason why to add it.

Comment (jppiiroinen) [reply github/License.py]: I guess I have misunderstood how the code works. I will do the requested change and remove RepositoryLicense class, as it does look a bit odd. I understood originally that there is an extra request which is done towards the License API itself without this change, but I admit that I have forgotten to verify this activity.

Comment (jppiiroinen) [reply github/License.py]: Let me know if there is something else which needs to be fixed.

**Comment by s-t-e-v-e-n-k** [comment]
@sfdye I think this is ready to merge, which option do I pick?

**Comment by jppiiroinen** [comment]
let me know if you would prefer to have a single commit in the branch, but as far as I remember the Github does mess up the review history when there is a rebase done for the branch itself.

I would prefer squash merge, as there are few WIP commits for those changes and those do not add anything for the actual change history on the master branch.

**Comment by s-t-e-v-e-n-k** [comment]
@jppiiroinen This is me still learning, I was given the commit bit for this repository a few hours ago. I would personally prefer a squash merge too, but I like how @sfdye does merges, and would like to do the same.

**Comment by jppiiroinen** [comment]
@s-t-e-v-e-n-k understood. :) ..just wanted to state my opinion while I was on the computer, and also let me know if you want me to change the history on the branch. It is not a problem to me to squash the commits myself and provide a single commit with a title and description for you.

**Comment by s-t-e-v-e-n-k** [comment]
@jppiiroinen Would you please rebase your change on top of https://github.com/PyGithub/PyGithub/pull/1155 and re-run the test suite? You don't have to push it up after doing so (and in fact, would make the diff hard to review if you do)

Sadly, due to a bug we weren't running the Repository tests, and your change directly impacts them.

**Comment by jppiiroinen** [comment]
> @jppiiroinen Would you please rebase your change on top of #1155 and re-run the test suite? You don't have to push it up after doing so (and in fact, would make the diff hard to review if you do)
> 
> Sadly, due to a bug we weren't running the Repository tests, and your change directly impacts them.

No worries. I will do so.

**Comment by jppiiroinen** [comment]
I have configured Travis CI for my personal fork and the rebased build is running here:
https://travis-ci.org/jppiiroinen/PyGithub/builds/551138341

**Comment by s-t-e-v-e-n-k** [comment]
That looks amazing! Except for the test failures. :-)

**Comment by jppiiroinen** [comment]
I have few minutes time right now to study why it failed. I will give the report here when it is ok.

**Comment by sfdye** [comment]
@s-t-e-v-e-n-k the default "Squash and merge" is good

**Comment by jppiiroinen** [comment]
To my eyes it looks to be an issue with some file under tests/ReplayData? are the actual requests mocks? So I need to update some of them as well to have the license response more likely.

I do not have more time right now and I will continue this later today

**Comment by s-t-e-v-e-n-k** [comment]
The replies in ReplayData are not mocks, but real responses from GitHub, usually with project names changed and so on -- it's what I suspected, which is that the repo returned in Repository.setUp.txt does not have the license field populated.

**Comment by s-t-e-v-e-n-k** [comment]
No stalebot, we want this when the tests pass.

**Comment by Humbertzhang** [comment]
> To my eyes it looks to be an issue with some file under tests/ReplayData? are the actual requests mocks? So I need to update some of them as well to have the license response more likely.
> 
> I do not have more time right now and I will continue this later today

Are there any new developments？

**Pull request closed without merging**